File: 5th sem/NM Computer Codes/gauss-jordan.c
#include<stdio.h>
#include<math.h>
#include<curses.h>

int main()
{
    int n, i, j, k;
    printf("\nEnter the number of variables: ");
    scanf("%d", &n);
    float a[n][n+1], r;
    printf("\nEnter the coefficients of the augmented matrix: \n");
    for (i = 1; i <= n; i++)
        for (j = 1; j <= n+1; j++)
            scanf("%f", &a[i][j]);
    for (j = 1; j <= n; j++)
    {
        if (fabs(a[j][j]) <= 0.0005)
        {
            printf("\nDiagonal element zero\n");
            return 1;
        }
        for (i = 1; i <= n; i++)
            if (i != j)
            {
                r = a[i][j] / a[j][j];
                for (k = 1; k <= n+1; k++)
                    a[i][k] -= r * a[j][k];
            }
    }
    printf("\nThe augmented diagonal matrix is: \n");
    for (i = 1; i <= n; i++)
    {
        for (j = 1; j <= n+1; j++)
            printf("%f\t", a[i][j]);
        printf("\n");
    }
    printf("\nThe solutions are:\n");
    for (i = 1; i <= n; i++)
        printf("x_%d = %f\n", i, a[i][n+1] / a[i][i]);
    return 0;
    getch();
}

File: 5th sem/NM Computer Codes/gauss_jordan.py
# gauss_jordan.py

n =int(input('Enter the number of variables: '))
a = []

print('Enter the coefficients')
for i in range(n):
    a.append([float(x) for x in input().split()])
    


print('matrix:')
for i in range(n):
    for j in range(n+1):
        print(a[i][j], end = ' ')
    print('\n')


for j in range(n):
    if abs(a[j][j]) < 0.005:
        print('Diagonal element Zero')
        exit()
    for i in range(n):
        if i != j:
            ratio = a[i][j] / a[j][j]
            for k in range(n+1):
                a[i][k] -= ratio * a[j][k]
                

print('Augmented Diagonal matrix:')
for i in range(n):
    for j in range(n+1):
        print(a[i][j], end = ' ')
    print('\n')
    
print('The solution is:')
for i in range(n):
    print('x_{0} = {1}\n'.format(i, a[i][n] / a[i][i]))
    
